csharp/src/Apache.Arrow/ArrowBuffer.cs

```diff
@@ -21,6 +21,10 @@ namespace Apache.Arrow
 {
     public readonly partial struct ArrowBuffer: IEquatable<ArrowBuffer>
     {
+        public static ArrowBuffer WrapExternalMemoryAsAnArrowBuffer(Memory<byte> externalMemory)
```

Comment:
I think we shouldn't be doing this, and instead just expose the existing constructor publicly. The constructor should continue taking `ReadOnlyMemory<byte>`.

Reply:
I'm good with that. I put in a new method here to make it super obvious that the memory is not owned by the buffer. The main concern was that making the constructor public would mean clients could construct an ArrowBuffer without going through the Builder. Given a choice, I'd go with forcing people to use the builder methods. They'd get the benefits of 64 byte aligned memory and everything else the builder methods do.

Reply:
@pgovind - what is the status of this comment? I think we should be exposing the constructor here, and not a static method.